akka-projection-core/src/main/scala/akka/projection/ProjectionSettings.scala

```diff
@@ -20,6 +20,8 @@ trait ProjectionSettings {
   def maxBackoff: FiniteDuration
   def randomFactor: Double
   def maxRestarts: Int
+  def saveOffsetAfterEnvelopes: Int
+  def saveOffsetAfterDuration: FiniteDuration
```

Comment:
I'm wondering if it would be better to no expose `ProjectionSettings`. 

Users can't change the values of `saveOffsetAfterEnvelopes` and `saveOffsetAfterDuration`, but they can see it even when not applicable (atMostOnce case). 

Then the only thing they can do, at least for the moment, is to programatically change the backoff settings. It feels strange that we have `withSaveOffset` and `withSettings`, instead of `withSaveOffset` and `withBackoff`.

Reply:
I'm not very comfortable with the nesting of settings, but if we were to continue the pattern then we might have something like:

```
.withSettings(...)
.withSaveOffsetSettings(SaveOffsetSettings(...))
.withBackoffSettings(BackoffSettings(...))
```

Which doesn't look great either.

Reply:
It would be nice to have all settings available at the first level like you suggested, but only the settings that are legal for that projection impl and message delivery type.

Reply:
I’m fine with moving backoff parameters to the top level and remove or hide the ProjectionSettings

Reply:
https://github.com/akka/akka-projection/issues/177